### ISSUE TITLE
Share by mail breaks File Sharing

### ISSUE THREAD

**Issue by FriedCircuits** [nextcloud/server#3457]
### Steps to reproduce
1. Enable share by mail
2. Try and share a folder 
3. Name resolution does not show and therefore can't not share

### Expected behaviour
When share by mail is disabled name lookup works and I am able to share a folder

### Actual behaviour
No name resolution so unable to share even if type name correctly. 

### Server configuration
**Operating system**:
Ubuntu Server 16.04
**Web server:**
Apache
**Database:**
MySQ
**PHP version:**
7
**Nextcloud version:** (see Nextcloud admin page)
11.0.1
**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from fresh 11
**Where did you install Nextcloud from:**
Nextcloud VM/Repos

**Are you using external storage, if yes which one:** local/smb/sftp/...
local
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Chrome
**Operating system:**
OSX/Windows 10
#### Browser log
<details>
<summary>Browser log</summary>

```
This is what the browser response is when it requests a search using the API

{"ocs":{"meta":{"status":"failure","statuscode":998,"message":"Invalid query, please check the syntax. API specifications are here: http://www.freedesktop.org/wiki/Specifications/open-collaboration-services. DEBUG OUTPUT:\n"},"data":[]}}
```
</details>

**Comment by nickvergessen** [comment]
Something in your nextcloud.log file?
What are you searching for? A local user?

**Comment by FriedCircuits** [comment]
I dont get anything in the Nextcloud log or apache logs. I could check again. Only clue was in the browser console.

**Comment by somoza** [comment]
Same here, Nextcloud 12 over Centos 7.

**Comment by c4539** [comment]
I have the same issue
My log is telling me the following:

> {"reqId":"******","level":3,"time":"2017-08-09T21:43:52+02:00","remoteAddr":"****","user":"****","app":"no app in context",
> "method":"GET","url":"\/ocs\/v1.php\/apps\/files_sharing\/api\/v1\/sharees?format=json&search=user&perPage=200&itemType=folder",
> "message":"Exception: {\"Exception\":\"InvalidArgumentException\",\"Message\":\"Invalid cloud id\",\"Code\":0,
> \"Trace\":\"#0 \\\/var\\\/www\\\/cloud.example.com\\\/apps\\\/files_sharing\\\/lib\\\/Controller\\\/ShareesAPIController.php(626): 
> OC\\\\Federation\\\\CloudIdManager->resolveCloudId('user@')\
> #1 \\\/var\\\/www\\\/cloud.example.com\\\/apps\\\/files_sharing\\\/lib\\\/Controller\\\/ShareesAPIController.php(553): OCA\\\\Files_Sharing\\\\Controller\\\\ShareesAPIController->getEmail('user')\
> #2 \\\/var\\\/www\\\/cloud.example.com\\\/apps\\\/files_sharing\\\/lib\\\/Controller\\\/ShareesAPIController.php(492): OCA\\\\Files_Sharing\\\\Controller\\\\ShareesAPIController->searchSharees('user', 'folder', Array, 1, 200, true)\
> #3 [internal function]: OCA\\\\Files_Sharing\\\\Controller\\\\ShareesAPIController->search('user', 'folder', 1, 200, NULL, true)\
> #4 \\\/var\\\/www\\\/cloud.example.com\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(160): call_user_func_array(Array, Array)\
> #5 \\\/var\\\/www\\\/cloud.example.com\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(90): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OCA\\\\Files_Sharing\\\\Controller\\\\ShareesAPIController), 'search')\
> #6 \\\/var\\\/www\\\/cloud.example.com\\\/lib\\\/private\\\/AppFramework\\\/App.php(114): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Files_Sharing\\\\Controller\\\\ShareesAPIController), 'search')\
> #7 \\\/var\\\/www\\\/cloud.example.com\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(47): OC\\\\AppFramework\\\\App::main('OCA\\\\\\\\Files_Shari...', 'search', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\
> #8 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\
> #9 \\\/var\\\/www\\\/cloud.example.com\\\/lib\\\/private\\\/Route\\\/Router.php(299): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\
> #10 \\\/var\\\/www\\\/cloud.example.com\\\/ocs\\\/v1.php(77): OC\\\\Route\\\\Router->match('\\\/ocsapp\\\/apps\\\/fi...')\\n#11 {main}\",\"File\":\"\\\/var\\\/www\\\/cloud.example.com\\\/lib\\\/private\\\/Federation\\\/CloudIdManager.php\",\"Line\":68}",
> "userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/60.0.3112.90 Safari\/537.36","version":"12.0.1.5"}

Reason for this is that the system address book contains some contacts with CLOUD = username@ instead of username@cloud.example.com.
I think this has something to do with legacy data from an earlier OwnCloud installation.
What do you thing?

**Comment by MorrisJobke** [comment]
Potential fix is in #6076

**Comment by c4539** [comment]
Much better than my quick fix. Thanks @nickvergessen !
